pkg/controller/template/render.go

```diff
@@ -197,23 +199,23 @@ func generateMachineConfigForRole(config *renderConfig, roleName string, roleDir
 		return nil, fmt.Errorf("error transpiling ct config to Ignition config: %v", err)
 	}
 
-	return machineConfigFromIgnConfig(roleName, ignCfg), nil
+	return machineConfigFromIgnConfig(role, name, ignCfg), nil
 }
 
 const (
 	machineConfigNameTmpl     = "00-%s"
 	machineConfigRoleLabelKey = "machineconfiguration.openshift.io/role"
 )
 
-func machineConfigFromIgnConfig(role string, ignCfg *ignv2_2types.Config) *mcfgv1.MachineConfig {
-	name := fmt.Sprintf(machineConfigNameTmpl, role)
```

Comment:
This will cause all machineconfigs to have `00-name`. We want to control the default (first in sorted order) machineconfig because, the ordering is important when appending ignition configs.

with current setup this would be possible..
```console
/worker/
    /00-worker/
/master/
    /00-master/
```

Reply:
fixed this up